.pre-commit-config.yaml

```diff
@@ -4,3 +4,24 @@ repos:
     hooks:
       - id: black
         language_version: python3
+  - repo: https://github.com/gitleaks/gitleaks
+    rev: v8.16.3
+    hooks:
+      - id: gitleaks
+  - repo: https://github.com/jumanjihouse/pre-commit-hooks
+    rev: 3.0.0
+    hooks:
+      - id: shellcheck
+  - repo: https://github.com/pocc/pre-commit-hooks
+    rev: v1.3.5
+    hooks:
+      - id: cpplint
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v4.4.0
+    hooks:
+      - id: end-of-file-fixer
+      - id: trailing-whitespace
```

Comment:
Out-of-the-box hooks from `pre-commit`, which already is a dependency. These particular hooks might save a lot of back-and-forth with CI regarding trailing whitespace errors. Approved :+1: